### PR TITLE
ci(dependabot): add dependabot config for updating dependencies

### PR DESCRIPTION
**Pull request by mohanraj-r**
### What does this PR do?
* Add config to enable dependabot to create PRs for updating dependencies 
* Add scheduled automated merging of minor/patch PRs

### What issues does this PR fix or reference?
@W-10247959@

### Functionality Before
* PRs to update dependencies were created manually

### Functionality After
* PRs to update dependencies are created automatically by dependabot and merged on schedule according to given config

### Discuss

- [x] Do we need to exclude any path (e.g. `/docs` or packages e.g. `@salesforce/*`, `@lwc/*` ?
- [x] Timing of opening PR (e.g. Tue following release cutoff)
- [x] Number of open PRs
- [x] Timing of PR merges (e.g. off-hours vs working hours) and number of PR merges per day
- [x] Major release PRs created against which branch

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -628,3 +629,20 @@ workflows:
...
           e: default-executor
+
+  dependabot-automerge:
+    triggers:
+      - schedule:
+          # TODO: change schedule before merge - currently set for freq testing
+          cron: '10 * * * *'
+          # Every hour during regular working hours (9am-5pm) 
+          # on Tue to Fri as Monday is release freeze.
+          # cron: '0 9-17 * * 2-5'
```
Comment (mohanraj-r) [inline .circleci/config.yml:640]: - 8pm to 6 am, wed-thurs
- check tz in circleci and adjust accordingly 
- start with 10 PRs per week ?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+# Must have a unique combination of 'package-ecosystem', 'directory', and
+# 'target-branch'. Hence different target branches are used for major vs
+# minor/patch updates.
+
+version: 2
+updates:
+  # Minor, Patch updates
+  - package-ecosystem: "npm"
```
Comment (mohanraj-r) [inline .github/dependabot.yml:13]: exclude /docs for now

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "daily"
```
Comment (mohanraj-r) [inline .github/dependabot.yml:16]: weekly

File: package.json [hunk 4]
```diff
@@ -64,10 +64,7 @@
     "link-lsp": "lerna exec yarn link @salesforce/aura-language-server @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lightning && lerna exec yarn link @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lwc",
     "unlink-lsp": "lerna exec yarn unlink @salesforce/aura-language-server @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lightning && lerna exec yarn unlink @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lwc"
   },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/forcedotcom/salesforcedx-vscode.git"
-  },
+  "repository": "mohanraj-r/salesforcedx-vscode",
```
Comment (mohanraj-r) [inline package.json:67]: forcedotcom

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    ignore:
+      - dependency-name: "*"
+        update-types:
+          ["version-update:semver-minor", "version-update:semver-patch"]
+      # TODO: Do we need to exclude specific packages to prevent W-7236739
+      #- dependency-name: "@lwc/*"
+      #- dependency-name: "@salesforce/*"
+    open-pull-requests-limit: 5
+    # TODO: Remove config before merging into origin to revert to default branch
+    target-branch: "dependabot-major-updates"
```
Comment (mohanraj-r) [inline .github/dependabot.yml:56]: remove the target branch for minor/patch updates and try to use develop here

**Review by mohanraj-r — COMMENTED** [review]
comments from walkthrough with team

File: docs/_articles/en/getting-started/recommended-extensions.md [hunk 6]
```diff
@@ -33,7 +33,7 @@ Analyzes your JavaScript code to find issues and to help you fix them. As a part
 
 ## XML
 
-VS Code doesn’t ship with rich XML tools by default. This extension, built by Red Hat, provides language support for for XML documents such as the `-meta.xml` or `package.xml` files in your VS Code project.
+VS Code doesn’t ship with rich XML tools by default. This extension, built by Red Hat, provides language support for XML documents such as the `-meta.xml` or `package.xml` files in your VS Code project.
```
Comment (mohanraj-r) [inline docs/_articles/en/getting-started/recommended-extensions.md:36]: minor typo cleanup

File: package.json [hunk 7]
```diff
@@ -64,10 +64,7 @@
     "link-lsp": "lerna exec yarn link @salesforce/aura-language-server @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lightning && lerna exec yarn link @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lwc",
     "unlink-lsp": "lerna exec yarn unlink @salesforce/aura-language-server @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lightning && lerna exec yarn unlink @salesforce/lwc-language-server @salesforce/lightning-lsp-common --scope salesforcedx-vscode-lwc"
   },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/forcedotcom/salesforcedx-vscode.git"
-  },
+  "repository": "forcedotcom/salesforcedx-vscode",
```
Comment (mohanraj-r) [inline package.json:67]: This is the format expected by release-management/dependabot-automerge
Apparently this is a valid format https://docs.npmjs.com/cli/v8/configuring-npm/package-json#repository
> For GitHub, GitHub gist, Bitbucket, or GitLab repositories you can use the same shortcut syntax you use for npm install

File: .github/dependabot.yml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      time: "06:00"
+      timezone: "America/Los_Angeles"
+    ignore:
+      - dependency-name: "*"
+        update-types:
+          ["version-update:semver-minor", "version-update:semver-patch"]
+    open-pull-requests-limit: 5
+  # Exclude updates to doc tools
+  # TODO: Enable after adding automated check(s) for doc generation
+  - package-ecosystem: "bundler"
```
Comment (mohanraj-r) [inline .github/dependabot.yml:42]: Also closed existing open PRs to `/docs` (about 4)

**Review by rcoringrato-sfdc — APPROVED** [review]

**Review by rcoringrato-sfdc — APPROVED** [review]

**Review by rcoringrato-sfdc — APPROVED** [review]

**Comment by mohanraj-r** [comment]
As we have configured circleci to not build PRs from forks (I had to create a fork to test out the circleci config), recreating this PR from a branch https://github.com/forcedotcom/salesforcedx-vscode/pull/3714

**Pull request closed without merging**